### DISCUSSION TITLE
what is .config/sww/wall.blur?

### DISCUSSION THREAD

**Discussion by ninfranko** [prasanthrangan/hyprdots#945] [Q&A] [answered]
i am trying to reverse engineer the rofi theme and am unable to add images to my config its just solid white and following the image location comes up as a dead end.

**Comment by kRHYME7** [comment]
this is a symbolic link

**Reply by ninfranko** [reply]
How does it work though

**Reply by kRHYME7** [reply, marked as answer]
It is used for backgrounds on rofi.

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/52cad804-2a6e-4833-9b47-2d1e9ee364c8)

``` ~/.config/hypr/scripts/swwwallpaper.sh ``` You can analyze this first on how this works. It's straightforward, so it will be easy. 

find the line ``` ln -fs "${cacheDir}/${curTheme}/${cacheImg}.blur" "${wallBlr}" ``` 
where ``` wallBlr="${XDG_CONFIG_HOME:-$HOME/.config}/swww/wall.blur" ``` 

+ This .blur file has the same name with ``` /${cacheImg} ``` but with .blur as suffix.  
+ This is the cached of the current wallpaper 
 + this blur version of the wallpaper is link to the ``` /.config/sww/wall.blur ``` 

With these lines you now realized how it works.
